capa/main.py

```diff
@@ -134,73 +123,13 @@ def set_vivisect_log_level(level):
     logging.getLogger("Elf").setLevel(level)
 
 
-def is_supported_format(sample: Path) -> bool:
-    """
-    Return if this is a supported file based on magic header values
-    """
-    taste = sample.open("rb").read(0x100)
-
-    return len(list(capa.features.extractors.common.extract_format(taste))) == 1
-
-
-def is_supported_arch(sample: Path) -> bool:
-    buf = sample.read_bytes()
-
-    return len(list(capa.features.extractors.common.extract_arch(buf))) == 1
-
-
-def get_arch(sample: Path) -> str:
-    buf = sample.read_bytes()
-
-    for feature, _ in capa.features.extractors.common.extract_arch(buf):
-        assert isinstance(feature.value, str)
-        return feature.value
-
-    return "unknown"
-
-
-def is_supported_os(sample: Path) -> bool:
-    buf = sample.read_bytes()
-
-    return len(list(capa.features.extractors.common.extract_os(buf))) == 1
-
-
-def get_os(sample: Path) -> str:
-    buf = sample.read_bytes()
-
-    for feature, _ in capa.features.extractors.common.extract_os(buf):
-        assert isinstance(feature.value, str)
-        return feature.value
-
-    return "unknown"
-
-
-def get_meta_str(vw):
-    """
-    Return workspace meta information string
-    """
-    meta = []
-    for k in ["Format", "Platform", "Architecture"]:
-        if k in vw.metadata:
-            meta.append(f"{k.lower()}: {vw.metadata[k]}")
-    return f"{', '.join(meta)}, number of functions: {len(vw.getFunctions())}"
-
-
-def is_running_standalone() -> bool:
-    """
-    are we running from a PyInstaller'd executable?
-    if so, then we'll be able to access `sys._MEIPASS` for the packaged resources.
-    """
-    return hasattr(sys, "frozen") and hasattr(sys, "_MEIPASS")
-
-
 def get_default_root() -> Path:
```

Comment:
should this function also go to helpers?

Reply:
yeah i was a little torn by this. i think only capa.main will ever run from PyInstaller, so it seemed like `is_running_standalone` should be in main, which means that `get_default_root` can't be in capa.helpers (circular import).

Reply:
I see, let's document that in the source and keep it as proposed.